core/io/resource.cpp

```diff
@@ -267,8 +267,11 @@ void Resource::reload_from_file() {
 }
 
 Variant Resource::_duplicate_recursive(const Variant &p_variant, const DuplicateParams &p_params, uint32_t p_usage) const {
-	// Anything other than object can be simply skipped in case of a shallow copy.
-	if (!p_params.deep && p_variant.get_type() != Variant::OBJECT) {
+	// Anything that doesn't isn't a container can be simply skipped in case of a shallow copy.
+	if (!p_params.deep && p_variant.get_type() != Variant::OBJECT && p_variant.get_type() != Variant::ARRAY && p_variant.get_type() != Variant::DICTIONARY) {
+		return p_variant;
+	}
+	if (p_usage & PROPERTY_USAGE_NEVER_DUPLICATE) {
 		return p_variant;
 	}
```

Comment:
Documentation of `PROPERTY_USAGE_NEVER_DUPLICATE` kinda implies this is for subresources 🤔 i know is not what it says, but is what i understood at first. Maybe it needs clarification?

<img width="575" height="91" alt="image" src="https://github.com/user-attachments/assets/2a8efa21-4349-4428-944d-2e0c5dd80888" />

Reply:
Yeah that would need updating.